saleor/graphql/account/mutations/staff.py

```diff
@@ -569,17 +570,24 @@ class Meta:
     def perform_mutation(cls, root, info: ResolveInfo, /, **data):
         user_id = data["user_id"]
         user = cls.get_node_or_error(info, user_id, field="user_id", only_type=User)
+        instance = cls.get_instance(info, **data)
+        data = data.get("input")
         with traced_atomic_transaction():
-            response = super().perform_mutation(root, info, **data)
-            if not response.errors:
-                manager = get_plugin_manager_promise(info.context).get()
-                address = manager.change_user_address(response.address, None, user)
-                remove_the_oldest_user_address_if_address_limit_is_reached(user)
-                user.addresses.add(address)
-                response.user = user
-                user.search_document = prepare_user_search_document_value(user)
-                user.save(update_fields=["search_document", "updated_at"])
-            return response
+            cleaned_input = cls.clean_input(info, instance, data)
```

Comment:
I am not convinced that copy&paste whole body of `ModelMutation.perform_mutation` is the best way to handle this. Can't you overwrite the base `clean_input` or `clean_instance` to call there a `super().clean_input/clean_instance` & `validation_address`?